### PR TITLE
Adding SD for superb (speech-classification).

### PR DESCRIPTION
**Pull request by Narsil**
- Adding a normalizers to help reformat the tensor into a list of classes
- Test for said normalizer (depends on torch, but it should be overall optional)
- Add the actual pipeline within superb.
- Will need update to `api-inference-community`.

### CODE REVIEW

File: api-inference-community/api_inference_community/normalizers.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+if TYPE_CHECKING:
+    try:
+        import torch
+    except Exception:
+        pass
+
+
+def speaker_diarization_normalize(
+    tensor: "torch.Tensor", sampling_rate: int, classe_names: List[str]
```
Comment (osanseviero) [inline api-inference-community/api_inference_community/normalizers.py:18]: nit: "classes_names" and same below

File: api-inference-community/docker_images/superb/app/pipelines/speech_classification.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import os
```
Comment (osanseviero) [inline api-inference-community/docker_images/superb/app/pipelines/speech_classification.py:1]: Few things
* Please add `speech-classification` to `common` as well.
* Could we also add a test with Lewis repo?

File: api-inference-community/docker_images/superb/app/pipelines/speech_classification.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    def __call__(self, inputs: np.array) -> List[Dict[str, Union[str, float]]]:
+        """
+        Args:
+            inputs (:obj:`np.array`):
+                The raw waveform of audio received. By default at 16KHz.
+                Check `app.validation` if a different sample rate is required
+                or if it depends on the model
+        Return:
+            A :obj:`dict`:. The object return should be liked {"text": "XXX"} containing
```
Comment (osanseviero) [inline api-inference-community/docker_images/superb/app/pipelines/speech_classification.py:47]: Please update the return type, this is still from ASR

**Review by osanseviero — COMMENTED** [review]
Overall looks great! Just a couple of comments

File: api-inference-community/api_inference_community/normalizers.py (hunk 1 above)
Comment (Narsil) [reply api-inference-community/api_inference_community/normalizers.py:18]: I really would like to avoid double ss in names, it's very confusing. 
-1 "s" at the end of a variables -> helps understand it's a list
- otherwise plurals are to be avoided IMO as it's very inconsistent for non natives (even for natives sometimes)

`class_names` or `classnames` if you prefer.

File: api-inference-community/docker_images/superb/app/pipelines/speech_classification.py (hunk 2 above)
Comment (Narsil) [reply api-inference-community/docker_images/superb/app/pipelines/speech_classification.py:1]: Sure !

File: api-inference-community/docker_images/superb/app/pipelines/speech_classification.py (hunk 3 above)
Comment (Narsil) [reply api-inference-community/docker_images/superb/app/pipelines/speech_classification.py]: Done.

File: api-inference-community/docker_images/superb/app/pipelines/speech_classification.py (hunk 2 above)
Comment (Narsil) [reply api-inference-community/docker_images/superb/app/pipelines/speech_classification.py:1]: Done.

File: api-inference-community/api_inference_community/normalizers.py (hunk 1 above)
Comment (Narsil) [reply api-inference-community/api_inference_community/normalizers.py]: I opted for `classnames` right now (classe is french so it was wrong anyway)

File: api-inference-community/docker_images/common/app/pipelines/speech_classification.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        )
+
+    def __call__(self, inputs: np.array) -> Dict[str, str]:
+        """
+        Args:
+            inputs (:obj:`np.array`):
+                The raw waveform of audio received. By default at self.sampling_rate, otherwise 16KHz.
+        Return:
+            A :obj:`list`:. Each item in the list is like {"class": "XXX", "start": float, "end": float}
+            "class" is the associated class of the audio segment, "start" and "end" are markers expressed in seconds
```
Comment (osanseviero) [inline api-inference-community/docker_images/common/app/pipelines/speech_classification.py:27]: Nit: maybe we should also mention that they can overlap (i.e. two speakers can speak at same time)

**Review by osanseviero — APPROVED** [review]
Thanks a lot! Approving this since it looks good IMO.

Last thing I think we should add are unit tests (under `superb/tests`)

File: api-inference-community/docker_images/common/app/pipelines/speech_classification.py (hunk 4 above)
Comment (Narsil) [reply api-inference-community/docker_images/common/app/pipelines/speech_classification.py:27]: If you feel that it is necessary, I'll add it (it's not necessarily speakers btw ^_^)

Comment (osanseviero) [reply api-inference-community/docker_images/common/app/pipelines/speech_classification.py]: Yes, speakers was just an example, it could be multiple classes in the same timeframe :)

File: api-inference-community/docker_images/common/app/pipelines/speech_segmentation.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    def __call__(self, inputs: np.array) -> Dict[str, str]:
+        """
+        Args:
+            inputs (:obj:`np.array`):
+                The raw waveform of audio received. By default at self.sampling_rate, otherwise 16KHz.
+        Return:
+            A :obj:`list`:. Each item in the list is like {"class": "XXX", "start": float, "end": float}
+            "class" is the associated class of the audio segment, "start" and "end" are markers expressed in seconds
+            within the audio file.
+        """
```
Comment (osanseviero) [inline api-inference-community/docker_images/common/app/pipelines/speech_segmentation.py:29]:
```suggestion
    def __call__(self, inputs: np.array) -> List[Dict[str, Union[str, float]]]:
        """
        Args:
            inputs (:obj:`np.array`):
                The raw waveform of audio received. By default at self.sampling_rate, otherwise 16KHz.
        Return:
            A :obj:`list`:. Each item in the list is like {"class": "XXX", "start": float, "end": float}
            "class" is the associated class of the audio segment, "start" and "end" are markers expressed in seconds
            within the audio file. Segments can overlap in any way.
        """
```

File: api-inference-community/tests/test_dockers.py [hunk 6]
```diff
@@ -223,6 +229,7 @@ def framework_docker_test(self, framework: str, task: str, model_id: str):
             self.assertEqual(response.content, b'{"ok":"ok"}')
 
             response = httpx.post(url, data=b"This is a test", timeout=timeout)
+            print(response.content)
```
Comment (osanseviero) [inline api-inference-community/tests/test_dockers.py:232]: Nit: remove?

**Review by osanseviero — APPROVED** [review]
Looks great!

File: api-inference-community/tests/test_dockers.py (hunk 6 above)
Comment (Narsil) [reply api-inference-community/tests/test_dockers.py:232]:
```suggestion
```

Comment (Narsil) [reply api-inference-community/tests/test_dockers.py:282]: Done.
